crypto/s2n_libcrypto.c

```diff
@@ -99,16 +99,33 @@ static S2N_RESULT s2n_libcrypto_validate_expected_version_number(void)
 }
 
 /* s2n_libcrypto_is_*() encodes the libcrypto version used at build-time.
- * Currently only captures AWS-LC and BoringSSL. When a libcrypto-dependent
- * branch is required, we prefer these functions where possible to reduce
- # #ifs and avoid potential bugs where the header containing the #define is not
- * included.
+ *
+ * When a libcrypto-dependent branch is required, we prefer these functions
+ * where possible to reduce #ifs and avoid potential bugs where the header
+ * containing the #define is not included.
  */
 
 #if defined(OPENSSL_IS_AWSLC) && defined(OPENSSL_IS_BORINGSSL)
     #error "Both OPENSSL_IS_AWSLC and OPENSSL_IS_BORINGSSL are defined at the same time!"
 #endif
 
+/* Attempt to detect if the libcrypto is OpenSSL.
+ *
+ * Since, quite a few libcrypto (BoringSSL, AWSLC) implementations are ABI
+ * compatible forks of OpenSSL, detecting OpenSSL is done by checking the
+ * absence of other libcrypto variants.
+ *
+ * Note: This check needs to be updated if s2n-tls adds support for a new
+ * libcrypto.
```

Comment:
I wonder if we actually could write a test to enforce this. Like add some string matching to https://github.com/aws/s2n-tls/blob/main/tests/unit/s2n_build_test.c.

Reply:
Awesome suggestion. I added a test for supported libcrytpo https://github.com/aws/s2n-tls/pull/4930/commits/6be1cfcef6eb106c83056b009cd7010fe557c13c